examples/discrete_turbines/tidal_array.py

```diff
@@ -0,0 +1,108 @@
+    t_end = 5*t_export
+
+# Bathymetry and viscosity sponges
+P1_2d = FunctionSpace(mesh2d, 'CG', 1)
+bathymetry_2d = Function(P1_2d, name='Bathymetry')
+bathymetry_2d.assign(Constant(50.0))
+x = SpatialCoordinate(mesh2d)
+h_viscosity = Function(P1_2d).interpolate(conditional(le(x[0], 50), 51-x[0], 1.0))
+File(outputdir + '/viscosity/viscosity.pvd').write(h_viscosity)
+
+
+# Turbine options
+turbine_thrust_def = 'table'  # 'table' or 'constant' (rated broken)
+include_support_structure = True
+speeds_AR2000 = [0., 0.75, 0.85, 0.95, 1., 3.05, 3.3, 3.55, 3.8, 4.05, 4.3, 4.55, 4.8, 5., 5.001, 5.05, 5.25, 5.5, 5.75,
+                 6.0, 6.25, 6.5, 6.75, 7.0]
+thrusts_AR2000 = [0.010531, 0.032281, 0.038951, 0.119951, 0.516484, 0.516484, 0.387856, 0.302601, 0.242037, 0.197252,
+                  0.16319, 0.136716, 0.115775, 0.102048, 0.060513, 0.005112, 0.00151, 0.00089, 0.000653, 0.000524,
+                  0.000442, 0.000384, 0.000341, 0.000308]
+turbine_density = Function(FunctionSpace(mesh2d, "CG", 1), name='turbine_density').assign(0.0)
+
+farm_options = DiscreteTidalTurbineFarmOptions()
+farm_options.turbine_type = turbine_thrust_def
+if turbine_thrust_def == 'table':
+    farm_options.turbine_options.thrust_speeds = speeds_AR2000
+    farm_options.turbine_options.thrust_coefficients = thrusts_AR2000
+else:
+    farm_options.turbine_options.thrust_coefficient = 0.6
+if include_support_structure:
+    farm_options.turbine_options.C_support = 0.7  # support structure thrust coefficient
+    farm_options.turbine_options.A_support = 3.5*10.0  # cross-sectional area of support structure
+    farm_options.turbine_options.diameter = 20
+farm_options.upwind_correction = False
+farm_options.turbine_density = turbine_density
+farm_options.turbine_coordinates = [[Constant(x), Constant(y)]
+                                    for x in numpy.arange(940, 1061, 60)
+                                    for y in numpy.arange(260, 341, 40)]
+
+
+# --- create solver ---
+solver_obj = solver2d.FlowSolver2d(mesh2d, bathymetry_2d)
+options = solver_obj.options
+options.simulation_export_time = t_export
+options.simulation_end_time = t_end
+options.output_directory = outputdir
+options.check_volume_conservation_2d = True
+options.fields_to_export = ['uv_2d', 'elev_2d']
+options.quadratic_drag_coefficient = Constant(0.0025)
+options.swe_timestepper_type = 'CrankNicolson'
+options.horizontal_viscosity = h_viscosity
+options.use_wetting_and_drying = True
+options.wetting_and_drying_alpha = Constant(0.5)
+options.discrete_tidal_turbine_farms[site_ID] = farm_options
+if not hasattr(options.swe_timestepper_options, 'use_automatic_timestep'):
+    options.timestep = 50.0
+
+options.swe_timestepper_options.solver_parameters = {
+    'snes_type': 'newtonls',
+    'snes_rtol': 1e-3,
+    'snes_linesearch_type': 'bt',
+    'snes_max_it': 20,
+    'ksp_type': 'preonly',
+    'pc_type': 'lu',
+    'pc_factor_mat_solver_type': 'mumps',
+}
```

Comment:
Are these needed? In examples I prefer to use default solver settings, or if needed explain why they are needed...